upup/pkg/fi/cloudup/awstasks/route.go

```diff
@@ -155,6 +155,15 @@ func (_ *Route) RenderAWS(t *awsup.AWSAPITarget, a, e, changes *Route) error {
 		} else if e.InternetGateway != nil {
 			request.GatewayId = checkNotNil(e.InternetGateway.ID)
 		} else if e.NatGateway != nil {
+			// It takes 'forever' (up to 5 min...) for a NatGateway to become available after it has been created
+			// We have to wait untill it is actualy up
+			params := &ec2.DescribeNatGatewaysInput{
+				NatGatewayIds: []*string{e.NatGateway.ID},
+			}
+			err := t.Cloud.EC2().WaitUntilNatGatewayAvailable(params)
```

Comment:
What is the default behavior of this API call? My concern is that we hit a problem, and the request takes *too long* to return.. Which seems very dangerous IMHO

Can we implement a timeout via the AWS API? Or a concurrent timeout via a goroutine/channel?